Amethyst/Managers/WindowManager.swift

```diff
@@ -29,33 +27,45 @@ enum MouseState {
     case dragging
     case moving(window: SIWindow)
     case resizing(screen: NSScreen, ratio: CGFloat)
-    case doneDragging(atTime: NSDate)
+    case doneDragging(atTime: Date)
+}
+
+protocol MouseStateKeeperDelegate: class {
+	func focusedScreenManager() -> ScreenManager?
+    func windows(on screen: NSScreen) -> [SIWindow]
+    func switchWindow(_ window: SIWindow, with otherWindow: SIWindow)
 }
 
 class MouseStateKeeper {
+    public let dragRaceThresholdSeconds = 0.15 // prevent race conditions during drag ops
     public var state: MouseState
-    private var _windowManager: WindowManager!
+    weak var mskDelegate: MouseStateKeeperDelegate!
     private var monitor: Any?
 
     init() {
         state = .pointing
     }
 
-    var windowManager: WindowManager {
+    deinit {
+        guard let oldMonitor = monitor else { return }
+        NSEvent.removeMonitor(oldMonitor)
+    }
+
+    var delegate: MouseStateKeeperDelegate {
```

Comment:
Do we still need to gate all of the setup on setting a delegate? It looks like none of the usage of the delegate is synchronously required.

Reply:
Aah, it looks like this logic can be simplified quite a bit.